homeassistant/components/ezviz/camera.py

```diff
@@ -0,0 +1,276 @@
+"""This component provides basic support for Ezviz IP cameras."""
+import asyncio
+import logging
+
+from haffmpeg.tools import IMAGE_JPEG, ImageFrame
+from pyezviz.client import EzvizClient, PyEzvizError
+import voluptuous as vol
+
+from homeassistant.components.camera import PLATFORM_SCHEMA, SUPPORT_STREAM, Camera
+from homeassistant.const import CONF_PASSWORD, CONF_USERNAME
+from homeassistant.helpers import config_validation as cv
+
+_LOGGER = logging.getLogger(__name__)
+
+CONF_CAMERAS = "cameras"
+CONF_SERIAL = "serial"
+CONF_DEFAULT_CAMERA_USERNAME = "admin"
+
+DATA_FFMPEG = "ffmpeg"
+
+CAMERAS_CONFIG = vol.Schema(
+    {
+        vol.Optional(CONF_USERNAME, default=CONF_DEFAULT_CAMERA_USERNAME): cv.string,
+        vol.Optional(CONF_PASSWORD): cv.string,
+        vol.Optional(CONF_SERIAL): cv.string,
+    }
+)
+
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
+    {
+        vol.Optional(CONF_CAMERAS, default={}): vol.All(),
```

Comment:
Something is missing within `vol.All()`.

Reply:
Changed to 'dict' check. Would you be able to help me validate this kind of configuration:
```
camera:
  - platform: ezviz
    username: 'em@il'
    password: 'pass'
    cameras:
      D12345678:
        username: admin
        password: ABCDEF
      D23456789:
        username: admin
        password: GHIJKL
      Dxxxxx:
...
```

the cameras sub-schema is a dict of key that are string, and are also dict of username/password that are strings

Reply:
Something like this:
```py
CAMERA_SCHEMA = vol.Schema(
    {
        CONF_USERNAME: cv.string,
        CONF_PASSWORD: cv.string,
    }
)

PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
    {
        CONF_USERNAME: cv.string,
        CONF_PASSWORD: cv.string,
        vol.Optional(CONF_CAMERAS, default={}):
            {
                cv.string: CAMERA_SCHEMA,
            },
    }
)
```

Reply:
I'm not sure about how you want to do optional vs required items so that may need to change. Specifically check when passing only the default empty dict.

Reply:
Thx!